project-koku/masu
# Synchronously removing expired report data. WIP #20

**Overview**
This change will synchronously remove expired report data when issuing a `DEL /api/v1/expired_data` when masu is in Debug mode.  This is to demonstrate that removal of report data is successful given the retention policy.  The default retention policy is 3 months of data and the policy can be configured with the environment variable `MASU_RETAIN_NUM_MONTHS`.

Issuing a `GET /api/v1/expired_data` will "simulate" the delete and provide a list of bills that will be removed should the `DELETE` command be executed.

**Testing**
Unit tests were added as well as the following functional tests:
1. Load May - July data from cost-usage-bucket
2. Hit GET /api/v1/expired_data and verify that no entries are returned with the default 3 month retention policy
3. Set env MASU_RETAIN_NUM_MONTHS to 2 and hit GET /api/v1/expired_data
4.  Hit DEL /api/v1/expired_data and verify data for May is actually removed


**Next Steps**
1. Add a new celery task to run at the beginning of each month to purge expired report data.
2. Optimize the database delete in `ReportDBCleaner().purge_expired_report_data()`

Functional Results:
[sync_remove_data_ut.txt](https://github.com/project-koku/masu/files/2243666/sync_remove_data_ut.txt)
